### PR TITLE
Fixing work_wires in `qml.PhaseAdder`

### PR DESCRIPTION
**Pull request by KetpuntoG**
Fixes a bug where `None` was added to the wires in `qml.PhaseAdder`, `qml.Adder` and `qml.OutAdder`.

This PR resolves this issue:

https://github.com/PennyLaneAI/pennylane/issues/6359

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -278,6 +278,8 @@
   [(#6278)](https://github.com/PennyLaneAI/pennylane/pull/6278)
   [(#6310)](https://github.com/PennyLaneAI/pennylane/pull/6310)
 
+* Fixes a bug where `None` was added to the wires in `qml.PhaseAdder`, `qml.Adder` and `qml.OutAdder`.
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:281]: The link is missing.

File: pennylane/templates/subroutines/out_adder.py [hunk 2]
```diff
@@ -205,11 +206,6 @@ def map_wires(self, wire_map: dict):
             new_dict["work_wires"],
         )
 
-    @property
```
Comment (soranjh) [inline pennylane/templates/subroutines/out_adder.py:208]: Why this is removed?

File: tests/templates/test_subroutines/test_adder.py [hunk 3]
```diff
@@ -204,6 +204,11 @@ def test_decomposition(self):
         for op1, op2 in zip(adder_decomposition, op_list):
             qml.assert_equal(op1, op2)
 
+    def test_work_wires_added_correctly(self):
+        """Test that no work wires are added if work_wire = None"""
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_adder.py:208]: How the template works with no work wires?

File: pennylane/templates/subroutines/out_adder.py [hunk 4]
```diff
@@ -205,11 +206,6 @@ def map_wires(self, wire_map: dict):
...
-    def wires(self):
-        """All wires involved in the operation."""
-        return self.hyperparameters["x_wires"] + self.hyperparameters["work_wires"]
-
```
Comment (willjmax) [inline pennylane/templates/subroutines/out_adder.py:212]: This should still return something. It just needs to check if there are work wires before returning.

File: pennylane/templates/subroutines/out_multiplier.py [hunk 5]
```diff
@@ -208,16 +214,6 @@ def map_wires(self, wire_map: dict):
...
-    @property
-    def wires(self):
-        """All wires involved in the operation."""
-        return (
-            self.hyperparameters["x_wires"]
-            + self.hyperparameters["y_wires"]
-            + self.hyperparameters["output_wires"]
-            + self.hyperparameters["work_wires"]
-        )
-
```
Comment (willjmax) [inline pennylane/templates/subroutines/out_multiplier.py:220]: Same as above, this should still be defined. It just needs to correctly handle the case where there are no work wires.

**Review by willjmax — COMMENTED** [review]
Looks good, just some minor suggestions

File: tests/templates/test_subroutines/test_adder.py (hunk 3 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_adder.py:208]: When the mod is an exact 2 power, no work qubits are needed.
It is in these cases where None was being included as wire 🐛

File: pennylane/templates/subroutines/out_multiplier.py (hunk 5 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/out_multiplier.py:220]: Safe to delete, after discussion

File: pennylane/templates/subroutines/out_adder.py (hunk 4 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/out_adder.py:212]: Safe to delete, after discussion

File: pennylane/templates/subroutines/out_adder.py (hunk 2 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/out_adder.py:208]: Safe to delete, after discussion

**Review by willjmax — APPROVED** [review]
Looks good :rocket:

File: pennylane/templates/subroutines/out_multiplier.py [hunk 6]
```diff
@@ -208,16 +216,6 @@ def map_wires(self, wire_map: dict):
...
 
-    @property
-    def wires(self):
-        """All wires involved in the operation."""
-        return (
-            self.hyperparameters["x_wires"]
-            + self.hyperparameters["y_wires"]
-            + self.hyperparameters["output_wires"]
-            + self.hyperparameters["work_wires"]
-        )
```
Comment (Jaybsoni) [inline pennylane/templates/subroutines/out_multiplier.py:219]: Yes, normally we just use the init method of the base class to set this property. There would have to be a special reason to override it here. Was there a specific reason this was added in the first place? If not its safe to remove 👍🏼

**Review by Jaybsoni — COMMENTED** [review]
Left a few small suggestions, otherwise looks good 💯

File: pennylane/templates/subroutines/out_adder.py [hunk 7]
```diff
@@ -171,10 +171,14 @@ def __init__(
             raise ValueError("None of the wires in y_wires should be included in output_wires.")
         for key in ["x_wires", "y_wires", "output_wires", "work_wires"]:
             self.hyperparameters[key] = qml.wires.Wires(locals()[key])
+
+        # pylint: disable=consider-using-generator
         all_wires = sum(
-            self.hyperparameters[key]
-            for key in ["x_wires", "y_wires", "output_wires", "work_wires"]
+            [self.hyperparameters[key] for key in ["x_wires", "y_wires", "output_wires"]], start=[]
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/out_adder.py:177]: Yes, if I follow pylint's suggestion, it includes the 0 as wire even if it is not part of the template

File: pennylane/templates/subroutines/out_multiplier.py (hunk 6 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/out_multiplier.py:219]: It was added recently and I do not remember any discussion about it

**Review by Jaybsoni — APPROVED** [review]
Looks good to me 👍🏼

**Pull request merged**